### PR TITLE
Type-Safe/Non-Boxing Global

### PR DESCRIPTION
**Pull request by martindevans**
Used the same mechanisms implemented for type-safe/non-boxing functions to wrap `Global` in the same way. I expected this to take longer to implement, but everything just fell into place perfectly 🎉

### CODE REVIEW

**Comment by martindevans** [comment]
Some open questions with how this is implemented:

 - Name: Currently it's `Global.Accessor<T>` but `Global.Wrapper<T>` may be more consistent?
 - Mutability: Currently `Set(T)` does a runtime check and throws if the global is immutable. It could be written to instead have `WrapGetter<T>` and WrapSetter<T>` methods, with `WrapSetter` returning null if the global is immutable.

File: src/Global.cs [hunk 1]
```diff
@@ -169,5 +184,72 @@ internal static class Native
         }
 
         private readonly ExternGlobal global;
+
+        /// <summary>
+        /// A typed accessor for a WebAssembly global value.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
```
Comment (peterhuene) [inline src/Global.cs:191]:
```suggestion
        /// <typeparam name="T">Type of the global being accessed.</typeparam>
```

**Comment by martindevans** [comment]
I've filled in that doc comment :)

**Comment by peterhuene** [comment]
Ignoring transient CI failure.

**Comment by peterhuene** [comment]
@martindevans thanks again for all these great contributions!

**Pull request merged**